src/hdusd/mx_nodes/nodes/base_node.py

```diff
@@ -434,26 +434,36 @@ def parse_val(prop_type, val, first_only=False):
 def create_node_types(file_paths):
     IGNORE_NODEDEF_DATA_TYPE = ('matrix33', 'matrix44')
 
-    node_def_classes = []
+    all_node_def_classes = []
+    node_def_cls_id = 0
+    all_mx_node_classes = []
+    mx_node_cls_id = 0
+
     for p in file_paths:
         doc = mx.createDocument()
         mx.readFromXmlFile(doc, str(p))
         mx_nodedefs = doc.getNodeDefs()
+        node_def_classes = []
         for mx_nodedef in mx_nodedefs:
             if MxNode._nodedef_data_type(mx_nodedef) in IGNORE_NODEDEF_DATA_TYPE:
                 continue
 
-            node_def_classes.append(MxNode.NodeDef.new(mx_nodedef))
+            node_def_classes.append(MxNode.NodeDef.new(mx_nodedef, node_def_cls_id))
+            node_def_cls_id += 1
+
+        # grouping node_def_classes by node and nodegroup
+        d = defaultdict(list)
```

Comment:
I think "nodes_definitions" or something like that would be more meaningful than "d".